msmtools/analysis/sparse/stationary_vector.py

```diff
@@ -119,7 +119,7 @@ def stationary_distribution_from_eigenvector(T, ncv=None):
 
     """
     vals, vecs = scipy.sparse.linalg.eigs(T.transpose(), k=1, which='LR', ncv=ncv)
-    nu = vecs[:, 0].real
+    nu = np.abs(vecs[:, 0].real)
```

Comment:
I recognized this just now, but are you sure that it is sound to use the abs here (instead of using the handling procedure for negative entries below)?

Reply:
I'm actually not convinced of either handling (old or new): I would vote for checking, if the sign structure is constant and raising otherwise (instead of just taking the abs).

Also I don't see a justification for taking the real (`vecs[:, 0].real`), since there shouldn't be any imaginary parts - at least in theory. Are there practical reasons for this?

Anyhow, I would additionally ensure (inside of the functions) that we are not trying to get a stationary distribution for a reducible matrix here, since this would lead to wrong results that could pass without complaint, if someone were to use the functions standalone.

Reply:
Okay, one by one:
1. check for constant sign structure and raise if not fulfilled - will implement
1. taking real part - there is no imaginary part, it's all zeros, yet, the format is `x + 0*y` for each entry in the vector. I suggest checking for imaginary part and raising if present
1. check for reducible matrix - will add a quick check for multiplicity of the 1-eigenvalue.